### PR TITLE
public link expiration date change acceptance test using webUI

### PR DESCRIPTION
**Pull request by iamsauravsharma**
## Description
test expiration date change of the public link using webUI

## How Has This Been Tested?
locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [X] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [X] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [X] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/WebUISharingContext.php [hunk 1]
```diff
@@ -322,6 +322,23 @@ public function theUserChangesThePermissionOfThePublicLinkNamedTo($name, $newPer
 		$this->addToListOfCreatedPublicLinks($name, $linkUrl);
 	}
 
+	/**
+	 * @When the user change the expiration of the public link named :linkName for file/folder :name to :date
```
Comment (paurakhsharma) [inline tests/acceptance/features/bootstrap/WebUISharingContext.php]:
```suggestion
	 * @When the user change the expiration of the public link named :linkName of file/folder :name to :date
```

File: tests/acceptance/features/bootstrap/WebUISharingContext.php [hunk 2]
```diff
@@ -433,6 +450,7 @@ public function theUserTriesToCreateANewPublicLinkForFileFolderUsingTheWebUIWith
 
 	/**
 	 * @Then the user should see an error message on the public link share dialog saying :expectedWarningMessage
+	 * @Then the user should see an error message saying :expectedWarningMessage
```
Comment (paurakhsharma) [inline tests/acceptance/features/bootstrap/WebUISharingContext.php]: I don't think this is necessary.

File: tests/acceptance/features/webUISharingExternal/shareByPublicLink.feature [hunk 3]
```diff
@@ -275,6 +275,29 @@ Feature: Share by public link
     Then the deleted elements should not be listed on the webUI
     And the deleted elements should not be listed on the webUI after a page reload
 
+  Scenario: user changes the expiration date of an already existing public link using webUI
+    Given user "user1" has created a share with settings
+      | path       | lorem.txt   |
+      | name       | Public link |
+      | expireDate | 14-10-2038  |
+      | shareType  | 3           |
+    When the user change the expiration of the public link named "Public link" for file "lorem.txt" to "21-07-2038"
```
Comment (paurakhsharma) [inline tests/acceptance/features/webUISharingExternal/shareByPublicLink.feature]:
```suggestion
    When the user change the expiration of the public link named "Public link" of file "lorem.txt" to "21-07-2038"
```

File: tests/acceptance/features/webUISharingExternal/shareByPublicLink.feature [hunk 4]
```diff
@@ -275,6 +275,29 @@ Feature: Share by public link
...
+    Then the fields of the last response should include
+      | expiration        | 21-07-2038              |
+
+  Scenario: user tries to change the expiration date of the public link to past date using webUI
+    Given user "user1" has created a share with settings
+      | path       | lorem.txt   |
+      | name       | Public link |
+      | expireDate | 14-10-2038  |
+      | shareType  | 3           |
+    When the user change the expiration of the public link named "Public link" for file "lorem.txt" to "14-09-2017"
```
Comment (paurakhsharma) [inline tests/acceptance/features/webUISharingExternal/shareByPublicLink.feature]:
```suggestion
    When the user change the expiration of the public link named "Public link" of file "lorem.txt" to "14-09-2017"
```

**Review by paurakhsharma — COMMENTED** [review]
Looks good :+1:

**Review by skshetry — COMMENTED** [review]
:+1:

File: tests/acceptance/features/bootstrap/WebUISharingContext.php [hunk 5]
```diff
@@ -322,6 +322,23 @@ public function theUserChangesThePermissionOfThePublicLinkNamedTo($name, $newPer
 		$this->addToListOfCreatedPublicLinks($name, $linkUrl);
 	}
 
+	/**
+	 * @When the user change the expiration of the public link named :linkName of file/folder :name to :date
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUISharingContext.php]: s/change/changes

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
@iamsauravsharma please backport

**Comment by iamsauravsharma** [comment]
backport on #34313

**Pull request merged**